lib/component-orchestrator/src/ComponentOrchestrator.js

```diff
@@ -356,104 +363,175 @@ class ComponentOrchestrator {
                 this._tokenSecret
             );
 
-            const flow = await this._flowsDao.findById(flowId);
+            const executionId = `msg-${flowExecId}-${stepId}-${v4()}`;
+            
+            const { isCompleted } = await this._stateMachine.initializeExecution({
```

Comment:
`isCompleted` will always be `false` here

`initializeExecution()` only returns `true` when it locates an item with the passed in `executionId`. But the `executionId` that is passed in is guaranteed to be unique, because it was generated two lines above as a string that includes a fresh uuid.

Reply:
I think this would need to be something like the message id so that we could know whether the message is being processed more than once.